### PR TITLE
chore: Updates TF to 1.9.6, Go to 1.23.1 and Go linter to 1.61.0

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Updates TF to 1.9.6, Go to 1.23.1 and Go linter to 1.61.0

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,2 +1,2 @@
-golang 1.23.0
-terraform 1.9.5
+golang 1.23.1
```
Comment (gssbzn) [inline .tool-versions:1]: given how the new toolchain works mixing with asdf can be cause issues so I'd recommend you use one or the other

**Review by gssbzn — APPROVED** [review]

File: .tool-versions (hunk 1 above)
Comment (lantoli) [reply .tool-versions:1]: the thing is that asdf needs a specific version, 1.23 won't work, but I think 1.23 (not 1.23.1) is a best practice in go.mod.

what do you suggest to do?

Comment (gssbzn) [reply .tool-versions:1]: in the CLI we turned off the toolchain install feature https://github.com/mongodb/mongodb-atlas-cli/blob/0b91d7e5eb3cd1da233394a8e9b35cb6cc1ba593/Makefile#L39

Comment (lantoli) [reply .tool-versions:1]: asdf is only used for local development as a "helper" for developers, but it's not used in any GH action for instance

Comment (gssbzn) [reply .tool-versions:1]: in the SDK I deleted the asdf go line

Comment (lantoli) [reply .tool-versions:1]: i see, if that's fine I'll keep it here as I think it's more handy for developers using asdf to have it

**Pull request merged**